src/AutoRest.CSharp/Common/Output/Models/ClassSignatureModifiers.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace AutoRest.CSharp.Common.Output.Models
+{
+    [Flags]
+    internal enum ClassSignatureModifiers
```

Comment:
As I mention in an above comment, I think we should call this as `TypeDeclarationModifiers` or `TypeSignatureModifiers`